packages/compliance-tests/src/topology/multicodec-topology.js

```diff
@@ -58,6 +57,9 @@ module.exports = (test) => {
         protocols: Array.from(topology.multicodecs)
       })
 
+      // 'change:protocols' event handler is async
+      await delay(10)
```

Comment:
why not listen for the event?

Reply:
The hander that invokes `topology._updatePeers` is async, so we can't just listen for the event as we're interested in the side-effects of the handler being invoked.

From a practical perspective the mock peer store used in the tests doesn't really do any async work, it just defers execution using the `async` keyword so `delay` here puts the rest of the test into the microtask queue - the handler is ahead of it in the same queue so this is enough for the test to pass.

Longer term we shouldn't test for side-effects like this as it's really fragile and makes refactoring hard.